packages/itwin/tree-widget/src/tree-widget-react/components/trees/models-tree/internal/ModelsTreeIdsCache.ts

```diff
@@ -272,19 +274,31 @@ export class ModelsTreeIdsCache {
       SELECT
         pe.ECInstanceId modeledElementId,
         pe.Category.Id categoryId,
-        pe.Model.Id modelId
+        pe.Model.Id modelId,
+        IFNULL((
+          SELECT false
+          FROM (
+            SELECT ECInstanceId id, IsPrivate isPrivate FROM BisCore.Model
+          )
+          WHERE id = pe.ECInstanceId AND isPrivate = false
+          LIMIT 1
+        ), true) isModeledElementPrivate
       FROM BisCore.Model m
       JOIN ${this._hierarchyConfig.elementClassSpecification} pe ON pe.ECInstanceId = m.ModeledElement.Id
+      JOIN ${this._hierarchyConfig.elementClassSpecification} ce ON ce.Model.Id = m.ECInstanceId
     `;
     for await (const row of this._queryExecutor.createQueryReader({ ecsql: query }, { rowFormat: "ECSqlPropertyNames", limit: "unbounded" })) {
-      yield { modelId: row.modelId, categoryId: row.categoryId, modeledElementId: row.modeledElementId };
+      yield { modelId: row.modelId, categoryId: row.categoryId, modeledElementId: row.modeledElementId, isPrivate: !!row.isModeledElementPrivate };
     }
   }
 
   private async getModelWithCategoryModeledElements() {
     this._modelWithCategoryModeledElements ??= (async () => {
       const modelWithCategoryModeledElements = new Map<Id64String, Id64Set>();
-      for await (const { modelId, categoryId, modeledElementId } of this.queryModeledElements()) {
+      for await (const { modelId, categoryId, modeledElementId, isPrivate } of this.queryModeledElements()) {
+        if (isPrivate) {
+          continue;
+        }
```

Comment:
we should filter out with query so we don't request unnecessary stuff from the backend just to drop it here